Algorithm.Python/CustomDataPropertiesRegressionAlgorithm.py

```diff
@@ -0,0 +1,136 @@
+﻿# QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+# Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from clr import AddReference
+AddReference("System.Core")
+AddReference("QuantConnect.Common")
+AddReference("QuantConnect.Algorithm")
+AddReference("QuantConnect.Algorithm.Framework")
+
+from System import *
+from QuantConnect import *
+from QuantConnect.Algorithm import *
+from QuantConnect.Algorithm.Framework import *
+from QuantConnect.Data import SubscriptionDataSource
+from QuantConnect.Python import PythonData
+from QuantConnect.Securities import *
+
+from datetime import datetime
+import json
+
+### <summary>
+### Regression test to demonstrate setting custom Symbol Properties and Market Hours for a custom data import
+### </summary>
+### <meta name="tag" content="using data" />
+### <meta name="tag" content="importing data" />
+### <meta name="tag" content="custom data" />
+### <meta name="tag" content="crypto" />
+### <meta name="tag" content="regression test" />
+class CustomDataPropertiesRegressionAlgorithm(QCAlgorithm):
+
+    def Initialize(self):
+        self.SetStartDate(2011,9,13)   # Set Start Date
+        self.SetEndDate(2015,12,1)     # Set End Date
+        self.SetCash(100000)           # Set Strategy Cash
+        
+        # Define our custom data properties and exchange hours
+        self.ticker = 'BTC'
+        properties = SymbolProperties("Bitcoin", "USD", 1, 0.01, 0.01, self.ticker)
+        exchangeHours = SecurityExchangeHours.AlwaysOpen(TimeZones.NewYork)
+
+        # Add the custom data to our algorithm with our custom properties and exchange hours
+        self.bitcoin = self.AddData(Bitcoin, self.ticker, properties, exchangeHours)
+
```

Comment:
Mind adding a line here for when its _not_ fully specified/using defaults; by memory, Python doesn't currently support optional parameters - just to make sure all the existing AddData implementations to break because it's not fully specified